kroxylicious-kms/src/main/java/io/kroxylicious/kms/service/Kms.java

```diff
@@ -45,7 +33,7 @@ public interface Kms<K, E> {
     CompletionStage<DekPair<E>> generateDekPair(@NonNull K kekRef);
```

Comment:
I'm not a fan of the `DekPair` name, does this give us the chance to drop that in favour of just calling it a `Dek`?

Reply:
I guess the risk is we are in quite a mathematical module with well-defined terms like dek and edek that will probably have some surrounding documentation later explaining envelope encryption. Does it muddy things if we have the record `Dek(SecretKey dek, E edek)`? It might make people think a dek is a thing with an edek inside it, rather than them being two representations of the same data.

On a similar note I wonder if we should change:
```
- CompletionStage<SecretKey> decryptEdek(@NonNull K kek, @NonNull E edek);
+ CompletionStage<DekPair<E>> decryptEdek(@NonNull K kek, @NonNull E edek);
```
so the decrypted thing has nicer names attached than just getting a SecretKey back

Reply:
Yeah, that makes sense as a rational we can't just call it `Dek`. 

`DekPair` -> `EncipheredDek` ?

Reply:
@SamBarker a `DekPair` is not _just_ an encrypted DEK either though. It's simply two representations of the same thing as @robobario says. It's also worth noting that [AWS uses the pair terminology](https://docs.aws.amazon.com/kms/latest/APIReference/API_GenerateDataKeyPair.html). (The other KMSs we've look at don't have to name the concept because they either rely on the DEK being generated by the application, and offer wrap and unwrap APIs, or in [the case of Hashicorp](https://developer.hashicorp.com/vault/api-docs/secret/transit#generate-data-key) use the term "data", which is clearly not helpful for us.) `DekPair` might not be the best name, but it's not terrible. `DekRepresentations`, for example, is much worse. `Deks` incorrectly implies multiple keys, etc.